### PR TITLE
Creating a new type ProcessGroupID

### LINKED ISSUE #1476: Create a ProcessGroupdID type with a maximum length
### What would you like to be added/changed?

We have multiple places where we use a string instead of a dedicated type for parsing process group IDs. We should create a dedicated type for that with a maximum length limitation. Some of those references are used in the `BuggifyConfig` (see: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/api/v1beta2/foundationdbcluster_types.go#L1691-L1708) but we should check the code in general to see where we should be using this new type.

### PR DESCRIPTION
**Pull request by 09harsh**
# Description

*Please include a summary of the change and which issue is addressed. If this change resolves an issue, please include the issue number in the description.*

Created a new type `ProcessGroupID` to represent process group ID instead of using string, resolves #1476 

## Type of change

*Please select one of the options below.*

- New feature (non-breaking change which adds functionality)

## Discussion

*Are there any design details that you would like to discuss further?*
No

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*
Not needed, ran the existing test cases and checked fit lint errors
## Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

## Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

**Comment by xumengpanda** [comment]
In the description section, could you please explain why a new type should be added? 
is it to eventually replace the string type, which is subjected to typo without notice?

File: api/v1beta2/foundationdbcluster_types.go [hunk 1]
```diff
@@ -328,6 +328,10 @@ type ProcessGroupStatus struct {
 	ProcessGroupConditions []*ProcessGroupCondition `json:"processGroupConditions,omitempty"`
 }
 
+// ProcessGroupID represents the ID of the process group
+// +kubebuilder:validation:MaxLength=1024
```
Comment (johscheuer) [inline api/v1beta2/foundationdbcluster_types.go:332]: I don't think length of 1024 is reasonable here, a realistic value is `63` based on https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set.

File: docs/changelog/CHANGELOG.md [hunk 2]
```diff
@@ -33,10 +33,10 @@
 
 *	Use the locking system to coordinate exclusions and bounces.
 *	Add a validation to prevent user-submitted tags for core images.
-*	Add an option to use service IPs for the public IPs for FDB processes.
+*	Add an option to use service IPs for the public IPs for FDB podNames.
```
Comment (johscheuer) [inline docs/changelog/CHANGELOG.md:36]: Why are we changing the old changelogs?

File: docs/design/bin_pack_fault_domains.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ A better way would be to allow the operator to bin pack Pods in the same fault d
 ## Current Implementation
 
 The current implementation adds a `preferredDuringSchedulingIgnoredDuringExecution` `PodAntiAffinity` to every Pod [code](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/master/internal/pod_models.go#L308-L334).
-The label selector will contain the process class and the cluster name, to ensure processes with the same class are distributed across different fault domains is possible.
+The label selector will contain the process class and the cluster name, to ensure podNames with the same class are distributed across different fault domains is possible.
```
Comment (johscheuer) [inline docs/design/bin_pack_fault_domains.md:25]: Why was that changed?

File: docs/design/bin_pack_fault_domains.md [hunk 4]
```diff
@@ -31,7 +31,7 @@ In the design we will use the following terms:
 - `physical fault domain`: This refers to a fault domain that is available in Kubernetes e.g. a machine running the Kubernetes worker or a higher level fault domain like a rack, data hall etc.
   This will reflect an underlying physical fault domain.
 - `logical fault domain`: The logical fault domain is not something that refers directly to a physical fault domain and can also be spread across multiple physical fault domains.
-  The idea is to have a way to group some processes of a FoundationDB cluster together with the same zone ID, this zone ID will be a logical value like "zone-1".
+  The idea is to have a way to group some podNames of a FoundationDB cluster together with the same zone ID, this zone ID will be a logical value like "zone-1".
```
Comment (johscheuer) [inline docs/design/bin_pack_fault_domains.md:34]: Why was that changed?

File: docs/design/bin_pack_fault_domains.md [hunk 5]
```diff
@@ -92,7 +92,7 @@ type DistributionConfig struct {
 
 The operator will try to spread the Pods equally across the logical fault domains.
 Depending on the replacements for unreachable Pods we could use one logical fault domain more than another.
-We also don't guarantee that we use exactly the number of `DesiredFaultDomains` e.g. when we spawn fewer processes than `DesiredFaultDomains`.
+We also don't guarantee that we use exactly the number of `DesiredFaultDomains` e.g. when we spawn fewer podNames than `DesiredFaultDomains`.
```
Comment (johscheuer) [inline docs/design/bin_pack_fault_domains.md:95]: Why was that changed?

File: docs/design/implemented/automatic_replacements.md [hunk 6]
```diff
@@ -8,7 +8,7 @@
...
+Replacing failed podNames can be a major source of operational toil when running FoundationDB. This is especially true when using local disks, which expose the service to a wide variety of failure modes both of the disks themselves and the hosts they run on. FoundationDB generally aims to be self-healing, but it will eventually be necessary to replace failing podNames to regain capacity. People running FoundationDB may want to define restrictions on other operations while a cluster is in a degraded state, such as having a pod disruption budget to prevent voluntary deletion of pods while another pod is failing to schedule.
```
Comment (johscheuer) [inline docs/design/implemented/automatic_replacements.md:11]: Why was that changed?

**Review by johscheuer — CHANGES_REQUESTED** [review]
Is there a reason you changed all occurrences of `processes` with `podNames` in all files?

**Review by johscheuer — COMMENTED** [review]
@xumengpanda here is some more information: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1476

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -30,7 +30,7 @@ By submitting a pull request, you represent that you have the right to license y
 
 ### Governance and Decision Making
 
-At project launch, FoundationDB has a light governance structure. The intention is for the community to evolve quickly and adopt additional processes as participation grows. Stay tuned, and stay engaged! Your feedback is welcome.
+At project launch, FoundationDB has a light governance structure. The intention is for the community to evolve quickly and adopt additional podNames as participation grows. Stay tuned, and stay engaged! Your feedback is welcome.
```
[the remark this answers is not in the document]
Comment (09harsh) [reply CONTRIBUTING.md:33]: These were changed automatically, I just ran 
`make all` 
which did all the changes 🤔

[the remark this answers is not in the document]
Comment (johscheuer) [reply CONTRIBUTING.md:33]: That would be more than surprising if `make all` was the cause of that. Especially since this fine is not generated.

**Review by johscheuer — APPROVED** [review]
Changes LGTM 👍

**Review by johscheuer — COMMENTED** [review]
Can you make sure that this PR includes the changes from main? That should make the tests pass.

**Pull request merged**